src/main/java/com/artipie/rpm/pkg/MetadataFile.java

```diff
@@ -46,7 +46,19 @@
  * and it proxies metadata to underlying output. After closing it saves
  * all metadata to {@code repomd.xml}.
  * @since 0.6
+ * @todo #99:30min MetadataFile should behave on close()
+ *  As stated in #99, MetadataFile should behave on close() method instead of
+ *  save(). Implement this and:
+ *  - remove MetadataFileTest.updatesRepomdOnSave
+ *  - enable MetadataFileTest.updatesRepomdOnClose
+ *  - remove AvoidDuplicateLiterals warning supression from MetadataFileTest
+ *  - remove save() method from MetadataFile
+ *  - remove suppressions from MetadataFile
+ *  - update coverage ratio after tests
```

Comment:
@paulodamaso honestly, I'm not sure I agree that `MetadataFile` should behave on `close()` instead of `save()`:
- we need to return path to saved metadata
- according to javadoc, `close()` method should close stream and release resources, not write anything to some other files
- close() method is called manually (not inside try-with-resources section)
- we are adding fields into `MetadataFile` that are needed only in one method, cohesion suffers

 So, I'd suggest getting rid of `close()` method and do not implement `Closable` interface for `Metadata`. WDYT? @g4s8 I'd like your opinion too.

Reply:
@olenagerasimova @paulodamaso it seems `close()` was initially designed for closing repomd but the behavior was changed during development, and it's not closing resources now. Maybe we can rename `close` to `save` (or something else) then?

Reply:
@g4s8 I still do not see why keep this `close()`: we can close `FileOutput` in `save()` and then write it to repomd. Now we have to remember, that `save()` should be called after `close()`, which does not make any sense to me.

Reply:
@olenagerasimova so can we remove it safely?

Reply:
@g4s8 yes, I do not see any difficulties here: move `this.out.close()` call to `save()` and delete `close()`

Reply:
@olenagerasimova @g4s8 I got confused, now. @victornoel suggested about moving the behavior from `save`  to `close`. I'm counting on the delete of `save` method. Essentially, moving `this.out.close()` to `save` and moving all `save` logic into `close` is the same thing. We won't have two methods anymore, just one.

Reply:
@paulodamaso was that what you had in mind maybe? I've read very quickly this PR, maybe I'm repeating what you already said ^^

Reply:
@g4s8 please, could you express your opinion here? What will we do?

Reply:
@olenagerasimova the proposed by @victornoel design sounds better to me, but I'm not sure if the implementation is still relevant.

Reply:
@olenagerasimova I agree with @victornoel proposed behavior; I prefer to have all work being done on close, trating it like a `Closeable` object, and remove `save` method.
@g4s8 What should we do?

Reply:
@paulodamaso please, be more practical, take a look at `Rpm#batchUpdate` and check how values, returned from `save`, are used. After that please suppose that `save` returns void. How will we put metadata to storage? 
(Current way is not correct, I understand, but I do not see how we can put metadata to storage is `save` will return nothing).

Reply:
@olenagerasimova 
> @paulodamaso please, be more practical, take a look at Rpm#batchUpdate and check how values, returned from save, are used. After that please suppose that save returns void. How will we put metadata to storage?
(Current way is not correct, I understand, but I do not see how we can put metadata to storage is save will return nothing).

It means changing the design of the object so that the creator of the object knows about the file being originally returned on save. So instead of having `save` create the `Path` that it returns, the Path should be passed to the constructor. From there, it will create a change of design that will impact the rest of the architecture until all those classes have much simpler interfaces with things happening on close and much declarative feeling in their use (IMHO).

Reply:
@victornoel 

> So instead of having save create the Path that it returns, the Path should be passed to the constructor.

I think that having fields that are used only in one method (and it will be so in this case) is not a good idea. We have `NamingPolicy` interface, I think it should be its job to provide `Path` to save metadata. And as this `NamingPolicy` is required only once on `save` it should be method parameter, not class field.

Reply:
> I think that having fields that are used only in one method (and it will be so in this case) is not a good idea

@olenagerasimova this is a design choice that I can't make as I am not ARC, but if you focus on what you just said, I think you are missing the point of the proposing design.

The point is the following I already shared:
> I had in mind a design similar to how output streams work in Java std lib: you create a new OutputStream(Path) and when you close it, you know that Path contains the desired information, but it's the responsibility of the stream creator to keep track of Path, not the one writing on the string or closing it. It is a kind of separation of preoccupation.

From this all the rest follows. But it's not the "best" solution, it's one design that seemed coherent with the rest of the design at the time the todo was written. Now maybe things are changed, but instead of focusing on just this class here, maybe the ARC team could take the time to highlight to the DEVs what is the global architecture that must be followed on this so that we all share the same vision? This could be by cleaning up the existing todo and creating new ones that better communicate this vision for example?